### DISCUSSION TITLE
SMTP Settings + Object Reference not set to an instance of an object

### DISCUSSION THREAD

**Discussion by wesfincher** [dnnsoftware/Dnn.Platform#5027] [Q&A] [answered]
Hey there - I'm not new to the DNN platform but relatively new in the community.  Hoping to become more active in the future.  For now, I could use some help...

Recently upgraded one of my sites to 9.10.1 (copied production to staging, upgraded site, updated host/portal settings via script).  I've done this 4 other times without any major issue.  This time I'm seeing an error in the persona bar when trying to access the SMTP settings.  Confirmed version 9.10.1 for Persona bar dlls
![image](https://user-images.githubusercontent.com/6493678/155588511-a65af873-c0f2-414b-84b5-9dcefc7a449a.png)

Here's the server response in the network tab of my browser:
{"Message":"An error has occurred.","ExceptionMessage":"Object reference not set to an instance of an object.","ExceptionType":"System.NullReferenceException","StackTrace":"   at Dnn.PersonaBar.Servers.Services.ServerSettingsSmtpHostController.GetSmtpSettings()"}

Looking at the code for ServerSettingsSmtpHostController (9.10.1), I'm not seeing any obvious culprits.  What other settings could I be missing?  I get this error on my server hosted site and when running my site locally.  How do I fix this?

**Comment by bdukes** [comment, marked as answer]
Welcome to the community 🎊

What version are you upgrading from?

Do you have a `<mail>` section in your `web.config` file? Is the `defaultProvider` attribute specified and matching an entry?

**Reply by wesfincher** [reply]
Thanks!  And thank you for chiming in...

I am upgrading from 9.0.2.  Neither my old nor my new web.config have a `<mail>` section in web.config.

**Reply by wesfincher** [reply]
This appears to have been the issue.  We were able to copy the `<mail>` section from another site and the SMTP settings now load.  Thank you!!

Follow up: If this was a failure in a portion of the upgrade, how do I easily identify what other portions of the upgrade may have failed?  Or how do I recover from a failed upgrade?

**Reply by bdukes** [reply]
Typically I would recommend reverting and doing the upgrade again if there were errors. Did you follow the [suggested upgrade path](https://docs.dnncommunity.org/content/getting-started/setup/upgrades/suggested-upgrade-path/index.html)?

To identify issues from the upgrade, you can look in `Portals/_default/Logs` for the `InstallerLogyyyyMMdd.resources` file(s) to see details of the package installations.

Hope it helps!
